src/themes/mixins.js

```diff
@@ -0,0 +1,174 @@
+// Let's start by generating a type scale, along with line-heights and small-caps modifiers.
```

Comment:
I'm gonna do a first pass on these functions, but for now as they're private (not exported) and we're very early on in building this, I'm not gonna be too picky about how they work. 😏

Reply:
Basically, there's a _lot_ of code in here and we should likely keep this file down to mixing we actually _export_. I'll want to add tests for a lot of these functions in the future as they are fundamental to how type gets rendered, but for now it largely seems good 👍

Reply:
Ideally, this file should stop being a junk drawer and we should have a type-styles.js file for the exports, and then maybe a type-utilities for *cough* type utilities like all the mathy stuff. I'm just being lazy and using it as a junk drawer right now. 

I'm also thinking we might want to refactor it so I can do 
`import { body } from 'themes/mixins';`

instead of

`import { bodySmall, bodyMedium, bodyLarge } from 'themes/mixins';`

...but we can handle that bit later. :)